### PR TITLE
Added tracing-opentelemetry-kafka

### PR DESCRIPTION
**Pull request by altro3**
Added tracing-opentelemetry-kafka.
Fixed tracing-opentelemetry-grpc packages.
Upgrade libs.
Fixed use deprecated code.

### CODE REVIEW

**Review by donbeave — APPROVED** [review]

File: .github/workflows/release.yml [hunk 1]
```diff
@@ -25,7 +25,7 @@ jobs:
           java-version: '17'
       - name: Set the current release version
         id: release_version
-        run: echo ::set-output name=release_version::${GITHUB_REF:17}
+        run: echo ::set-output name=release_version::${GITHUB_REF:11}
```
[the remark this answers is not in the document]
Comment (altro3) [reply .github/workflows/release.yml:28]: No, it isn't. See https://github.com/micronaut-projects/micronaut-core/pull/8203

File: tracing-bom/build.gradle [hunk 2]
```diff
@@ -35,6 +44,7 @@ micronautBom {
         dependencies.add("io.zipkin.reporter2:zipkin-reporter-bom:2.16.3")
         dependencies.add("io.opentelemetry:opentelemetry-bom:1.19.0")
         dependencies.add("io.opentelemetry:opentelemetry-bom-alpha:1.19.0-alpha")
-        dependencies.add("io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom:1.19.0")
+        dependencies.add("io.opentelemetry.instrumentation:opentelemetry-instrumentation-bom:1.19.1")
```
[the remark this answers is not in the document]
Comment (altro3) [reply tracing-bom/build.gradle:47]: @n0tl3ss i don't know. This "hack" was added by @graemerocher

File: tracing-opentelemetry/src/main/java/io/micronaut/tracing/opentelemetry/DefaultOpenTelemetryFactory.java [hunk 3]
```diff
@@ -79,28 +85,33 @@ protected OpenTelemetry defaultOpenTelemetry(ApplicationConfiguration applicatio
...
+                    if (idGenerator != null) {
+                        tracerProviderBuilder.setIdGenerator(idGenerator);
+                    }
+                    if (spanProcessor != null) {
+                        tracerProviderBuilder.addSpanProcessor(spanProcessor);
+                    }
+                    if (resourceProvider != null) {
+                        tracerProviderBuilder.setResource(resourceProvider.resource());
+                    }
+                    if (applicationConfiguration.getName().isPresent()) {
```
[the remark this answers is not in the document]
Comment (altro3) [reply tracing-opentelemetry/src/main/java/io/micronaut/tracing/opentelemetry/DefaultOpenTelemetryFactory.java]: @n0tl3ss fixed

**Review by n0tl3ss — APPROVED** [review]

**Comment by graemerocher** [comment]
Needs documentation

**Comment by altro3** [comment]
@graemerocher After last changes can't add `api mn.micronaut.core.processor` dependency
изображение

**Comment by altro3** [comment]
@graemerocher after last changes have this error:
изображение

**Comment by altro3** [comment]
@graemerocher @donbeave @n0tl3ss rebase these changes to version 4.5.0. See https://github.com/micronaut-projects/micronaut-tracing/pull/200

**Pull request closed without merging**